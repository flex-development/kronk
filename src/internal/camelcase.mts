/**
 * @file Internal - camelcase
 * @module kronk/internal/camelcase
 */

/**
 * Convert a kebab-case `string` to camelCase.
 *
 * @internal
 *
 * @this {void}
 *
 * @param {string} string
 *  The string to convert
 * @return {string}
 *  Camel cased `string`
 */
function camelcase(this: void, string: string): string {
  return string.split('-').reduce((acc, part) => {
    return acc + part[0]!.toUpperCase() + part.slice(1)
  })
}

export default camelcase
