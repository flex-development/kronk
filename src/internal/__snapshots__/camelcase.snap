// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:internal/camelcase > should return camelCased string ("debug") 1`] = `"debug"`;

exports[`unit:internal/camelcase > should return camelCased string ("esbuild.log-level") 1`] = `"esbuild.logLevel"`;

exports[`unit:internal/camelcase > should return camelCased string ("logLevel") 1`] = `"logLevel"`;

exports[`unit:internal/camelcase > should return camelCased string ("pizza-type") 1`] = `"pizzaType"`;

exports[`unit:internal/camelcase > should return camelCased string ("resolve.conditions") 1`] = `"resolve.conditions"`;
